### DISCUSSION TITLE
More NES dump problems - different CHR & PRG every time

### DISCUSSION THREAD

**Discussion by muguemjee** [sanni/cartreader#283] [Need help]
Im experiencing more problems with the sanni for dumping nes cartridges. For some reason its dumping files that are never the same. Prg and chr keeps changing with every dump. I verified the switches are good and everything same settings each time

**Comment by Dakkaron** [comment]
Have you compared the hex dump of your dumps and ROMs you find online? Maybe there is a pattern in the reading errors.

Did you check all your soldering points? Did you clean your NES cartridge contacts with alcohol or contact spray?

**Reply by jon-nielsen** [reply]
Also make sure to set RAM amount even if just dumping the ROMs

**Reply by muguemjee** [reply]
While i havent compared the rom dumps are different from each other on every attempt. Even for the same cart. It even happens when i dump the whole cart as well as individually PRG and CHR. Sometimes in these cases the prg.bin matches its corresponding portion of data in the cart.bin while the chr.bin is different. And then if i dump the cart again its the inverse. And all the while still being different from the previous attempt files

Ill try cleaning the cartridge contacts. I forgot to mention that snes cartridge dumping works fine.

How much should i set the ram to? All the games have 0 by default according to the nes card db

**Reply by Dakkaron** [reply]
The reason I was asking about the ROM is that sometimes there are patterns which can point you into the right direction. E.g. there could be a loose contact on a pin or two, and then you will see that always the same bits are wrong.

**Reply by muguemjee** [reply]
Ok. Yeah in this case the dump files are never consistent across from each other. The locations in the file where they differ also differ from each other each time. That snes cartridges are dumping fine, and prototypes even. Makes me think maybe its something with the nes adapter or maybe the nes cartridges although it did this for 4 nes cartridges not one dumped correctly

**Comment by muguemjee** [comment]
Does it support dumping nes prototypes? These are what im trying to dump specifically

**Reply by jon-nielsen** [reply]
Well if it actually is 0 then it's correct to set that. You say it's prototypes, so there's the possibility that Nescartdb etc may not match mapper or ROM sizes exactly for what you have. There's also the possibility of the cart simply not working right (hopefully not the case!). I'd probably open up the cart and inspect PCB and components

**Reply by muguemjee** [reply]
That would be unfortunate if theyre defective i hope thats not the case. But i kind of doubt this because 4 prototypes were attempted and theyre all having this same type of problem

But it does look like the only thing to do next is open up the cartridge. Need to make sure the mapper and sizes dont differ. Im not familiar with inspecting the pcb is this information labelled on the board somewhere? Although this possibility im also doubtful. Not just because the same issue is happening to 4 different carts

**Reply by jon-nielsen** [reply]
Well maybe it's early to assume the carts are bad. You say that SNES carts dump fine, so it's probably not bad soldering of the cart reader. Did you try non-proto NES carts? I do have some experience with the NES adapter being harder to make good contact.

**Reply by muguemjee** [reply]
i should definitely give that a go trying some retail carts. A lot of possibilities to look into. Hopefully one of these things pan out.

**Comment by Androxilogin** [comment]
I'd like to add that I assembled this converter and cart reader in question
and it passed all tests/dumped retail NES carts to the correct file
extension before shipping it out. I'm unfamiliar with a lot of the NES
stuff though so beyond the basics I can't really answer for.

**Reply by muguemjee** [reply]
Do you know or recall what mappers your carts have?

**Reply by Androxilogin** [reply]
I dumped Double Dragon II (mapper 04), Ice Hockey (mapper 00), NFL (mapper 02) and also Q*Bert (mapper 03).

**Reply by muguemjee** [reply]
Ok. Well thats ruled out at least. I got a pattern where all the prototypes dumped were on mapper 4 so i thought it might have had to do with the script written for that being finnicky. But since double dragon 2 works when you dumped it it cant be related to that mapper

**Reply by Androxilogin** [reply]
I was just assembling one of these and it failed the test. Two pins were getting continuity, I checked them on their solder pads and everything looked fine. Looked at the pins and found that two were pushed in a bit further than they should have been. Not sure if there's a bottom rail here that would be in easy reach of contact but figured I would mention. Brought them back up with a dentist pik and it started functioning properly. It was hard to catch, but **[here is a pic](https://imgur.com/a/Uubef8B)** along with the pinout diagram. Unimportant, but mine happened to be pins 10 & 11.

**Reply by muguemjee** [reply]
Interesting. Thats going to be hard to catch in the current situation but i should look into that. Did you test any nes carts before noticing the pins pushed in? If so did it do something similar?

**Comment by Androxilogin** [comment]
Yeah, same result. Pins were pushed in, it dumped as a .bin file. After the
fix it output as .nes

**Reply by muguemjee** [reply]
Definitely have to check that. In this case the files are being output as bin instead of nes so somethings off somewhere no doubt.
